object_store/src/lib.rs

```diff
@@ -231,6 +231,20 @@ pub trait ObjectStore: std::fmt::Display + Send + Sync + Debug + 'static {
     /// in the given byte range
     async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes>;
 
+    /// Return the bytes that are stored at the specified location
+    /// in the given byte ranges
+    async fn get_ranges(
+        &self,
+        location: &Path,
+        ranges: &[Range<usize>],
+    ) -> Result<Vec<Bytes>> {
+        let mut out = Vec::with_capacity(ranges.len());
+        for range in ranges {
+            out.push(self.get_range(location, range.clone()).await?)
+        }
+        Ok(out)
```

Comment:
Why not use  `coalsce_ranges` here as the default implementation? That would likely perform better for user supplied object store implementations, and this PR could likely get smaller too as AWS, GCP, etc would not have to be modified

```suggestion
        coalesce_ranges(
            ranges,
            |range| self.get_range(location, range),
            OBJECT_STORE_COALESCE_DEFAULT,
        )
        .await
```

Reply:
Yea, `coalsce_ranges` looks should outperform current default implementation. Looks good to use it as default one.

Reply:
I guess I was concerned that some implementations may not benefit from this, but then again perhaps those cases should override this 🤔